apps/api/views/metrics.py

```diff
@@ -0,0 +1,750 @@
+            output_list.append(l2[j])
+            if j < len(l2) - 1:
+                j += 1
+            else:
+                l2_end = True
+
+        elif l1[i][union_key] == l2[j][union_key]:
+            l1[i].update(l2[j])
+            output_list.append(l1[i])
+            if j < len(l2) - 1:
+                j += 1
+            else:
+                l2_end = True
+            if i < len(l1) - 1:
+                i += 1
+            else:
+                l1_end = True
+
+    return output_list
+
+
+class InstructorOrSuperuserPermission(permissions.BasePermission):
+    message = 'You are not allowed to access this data.'
+
+    def has_permission(self, request, view):
+        if request.user.is_superuser:
+            return True
+
+        klass_pk = view.kwargs.get('klass_pk')
+        student_pk = view.kwargs.get('student_pk')
+        team_pk = view.kwargs.get('team_pk')
+
+        instructor = None
+        try:
+            instructor = request.user.instructor
+        except AttributeError:
+            return False
+
+        if klass_pk:
+            klass = get_object_or_404(Klass, pk=klass_pk)
+            if instructor == klass.instructor:
+                return True
+
+        elif student_pk:
+            student = get_object_or_404(StudentMembership, pk=student_pk)
+            if instructor == student.klass.instructor:
+                return True
+
+        elif team_pk:
+            team = get_object_or_404(Team, pk=team_pk)
+            if instructor == team.klass.instructor:
+                return True
+        return False
+
+
+class TimeDistributionMixin:
+    # Empty Default values
+    time_distribution_model = None
+    time_distribution_kwarg_name = None
+    time_distribution_filter_name = None
+
+    def time_distribution_query(self, **kwargs):
+        kwarg = kwargs.get(self.time_distribution_kwarg_name)
+        filter = {
+            self.time_distribution_filter_name: kwarg,
+        }
+        data = self.time_distribution_model.objects.filter(**filter).extra({'time': "EXTRACT(HOUR FROM created)"}).values('time').order_by('time').annotate(count=Count('pk'))
+        return data
+
+
+class ScoreDistributionMixin:
+    def score_distribution_query(self):
+        sub_sample = Submission.objects.filter(tracked_submissions__stored_score__isnull=False).order_by(
+            '?').values(score=(F('tracked_submissions__stored_score') - F('definition__baseline_score')) / (
+                F('definition__target_score') - F('definition__baseline_score')))[:1000]
+        sorted_sample = sorted(list(sub_sample), key=lambda k: k['score'])
+
+        # Histogram settings
+        minimum = 0
+        maximum = 1.2
+        delta = maximum - minimum
+        bucket_quantity = 12
+        bucket_bounds = []
+
+        for i in range(bucket_quantity + 1):
+            bucket_bound = round(minimum + (i * delta / bucket_quantity), 1)
+            bucket_bounds.append(bucket_bound)
+
+        i = 0
```

Comment:
Hmmm, I would say same comment as above as far as descriptive variable names, but this may make sense in this case for general iterators...

Reply:
It seems that in the context of creating a histogram that bucket bounds would make sense. Do you have a recommendation for increasing the clarity here? I left a small comment for clarification.

Reply:
I don't really have any good suggestions for this I guess. It should just be known that i/j are index's for the current iterator I guess.

Reply:
I'm pickin' up what you're puttin' down. Done.